packages/flutter/lib/src/services/deferred_component.dart

```diff
@@ -0,0 +1,157 @@
+/// feature is ready to use or not. Readiness should be determined by
+/// completion of the future returned by
+/// [DeferredComponent.installDeferredComponent] or `loadLibrary()`.
+///
+/// These states are an extension of the states in Android's
+/// https://developer.android.com/reference/com/google/android/play/core/splitinstall/model/SplitInstallSessionStatus
+///
+/// Typical state flow begins as `unknown`, and transitions into `requested`
+/// when the installation is begun. When the request is processed, the
+/// state changes to `downloading` and finally `installed`. Modules previously
+/// installed but not loaded in this session will be in the
+/// `installedPendingLoad` state, which indicates that the installation request
+/// call (either `loadLibrary()` or [DeferredComponent.installDeferredComponent]) should
+/// be repeated to trigger the loading process.
+enum DeferredComponentInstallState {
+  /// The deferred component installation has been requested, but has not begun
+  /// download or installation yet.
+  requested,
+  /// The deferred component installation is awaiting further action.
+  pending,
+  /// User input is requried for continued installation.
+  requireUserConfirmation,
+  /// The deferred component is being downloaded.
+  downloading,
+  /// The deferred component has finished downloading but has not yet been
+  /// installed.
+  downloaded,
+  /// The downloaded deferred component is being installed.
+  installing,
+  /// The deferred component has previously been installed and all files/assets
+  /// are available, but has not been loaded into the VM in the current
+  /// app's session. Either `loadLibrary()` or [DeferredComponent.installDeferredComponent]
+  /// should be called to trigger the loading process to ensure the deferred component's
+  /// components can be safely used in the app.
+  installedPendingLoad,
+  /// The deferred component has been succesfully downloaded and installed. Assets and
+  /// code from the deferred component are ready to use.
+  installed,
+  /// An installation request has been asked to be canceled.
+  cancelling,
+  /// An installation request has been canceled.
+  canceled,
+  /// An installation request has failed.
+  failed,
+  /// The deferred component has not been requested nor installed or is unavailable.
+  unknown,
+}
+
+/// Manages the installation and loading of deferred component modules.
+///
+/// Deferred components allow Flutter applications to download precompiled AOT
+/// dart code and assets at runtime, reducing the install size of apps and
+/// avoiding installing unnessecary code/assets on end user devices. Common
+/// use cases include deferring installation of advanced or infrequently
+/// used features and limiting locale specific features to users of matching
+/// locales.
+class DeferredComponent {
+  // This class is not meant to be instantiated or extended; this constructor
+  // prevents instantiation and extension.
+  // ignore: unused_element
```

Comment:
nit: I think this ignore is no longer needed, the "unused_element" has been fixed to recognize that this constructor is in-fact used (to prevent instantiation), I believe.